tox.ini

```diff
@@ -11,10 +11,10 @@ deps =
     portend
     pytest-services
     munch
+passenv =
+    PYTEST_ADDOPTS
 commands =
-    pytest --cov=twine --cov-context=test \
-        --cov-report term-missing --cov-report html \
-        {posargs:tests}
+    pytest {posargs:--cov=twine --cov-context=test --cov-report term-missing --cov-report html tests}
```

Comment:
I've never liked this form, where lots of default arguments are passed if no arguments are passed... because then passing any other argument, related or unrelated, will disable all of the defaults. Ideally, supplying a parameter such as `-k integration` would not disable coverage (or potentially other options that get added here later). That's why I put default arguments for coverage in .coverage or pytest.ini. That said, this usage is acceptable.

Reply:
Yeah, I was trying to strike a balance between running the whole test suite with coverage, and running a subset of tests. In the latter case, I generally care less about coverage, and find the text report distracting.

I hadn't thought about putting them in `pytest.ini`; I'll take a look at that. Off the top of my head, it seems like `--cov` and `--cov-context` could go there, while `--cov-report` could stay here.

As I noted [earlier](https://github.com/pypa/twine/pull/633#discussion_r429537784), another option might be to have a separate `coverage` testenv, e.g.:

https://github.com/sphinx-doc/sphinx/blob/master/tox.ini
https://github.com/jazzband/pip-tools/blob/master/tox.ini

Are there any cons to that approach?

Reply:
I moved some options to `pytest.ini` in https://github.com/pypa/twine/pull/633/commits/22ece54f718a55449022a3eaf659c4696f6f5b28.